src/analysis/scoping_analysis.h

```diff
@@ -62,12 +63,24 @@ class ScopeInfo {
     //  print dis.dis(g)
 
     enum class VarScopeType { FAST, GLOBAL, CLOSURE, DEREF, NAME };
-
-    virtual bool refersToGlobal(InternedString name) = 0;
-    virtual bool refersToClosure(InternedString name) = 0;
-    virtual bool saveInClosure(InternedString name) = 0;
     virtual VarScopeType getScopeTypeOfName(InternedString name) = 0;
 
+    // Returns true if the variable should be passed via a closure to this scope.
+    // Note that:
+    //      (a) This can be false even if there is an entry in the closure object
+    //          passed to the scope, if the variable is not actually used in this
+    //          scope or any child scopes.
+    //      (b) This can be true even if it is not used in this scope, if it
+    //          is used in a child scope.
+
+    virtual bool isPassedToViaClosure(InternedString name) = 0;
```

Comment:
Could this comment say something about why this function exists (that these names show up in locals even though they might never be referenced in the block)?  It's hard to look at this description and understand how exactly this is different from the other functions -- ex would this be the same as `getScopeTypeOfName() == DEREF || CLOSURE`?  Saying that it's used for a different purpose makes it more clear that we're not necessarily exposing different functionality here, just serving a slightly different purpose.  (Similarly it might be more clear to call it something like `closureNameShowsInLocals`.)